gobblin-service/src/main/java/org/apache/gobblin/service/modules/dataset/FSDatasetDescriptor.java

```diff
@@ -113,41 +119,61 @@ protected boolean isPathContaining(DatasetDescriptor other) {
    * accepted by the other {@link DatasetDescriptor}. If the path description of the other {@link DatasetDescriptor}
    * is a glob pattern, we return false.
    *
-   * @param otherPath a glob pattern that describes a set of paths.
+   * @param userFlowConfigPath a glob pattern that describes a set of paths.
    * @return true if the glob pattern described by the otherPath matches the path in this {@link DatasetDescriptor}.
    */
-  private boolean isPathContaining(String otherPath) {
-    if (otherPath == null) {
-      return false;
+  private ArrayList<String> isPathContaining(String userFlowConfigPath, Boolean inputDataset) {
+    String datasetDescriptorPrefix = inputDataset ? DatasetDescriptorConfigKeys.FLOW_INPUT_DATASET_DESCRIPTOR_PREFIX : DatasetDescriptorConfigKeys.FLOW_OUTPUT_DATASET_DESCRIPTOR_PREFIX;
+    ArrayList<String> errors = new ArrayList<>();
+    if (userFlowConfigPath == null) {
+      errors.add(datasetDescriptorPrefix + DatasetDescriptorConfigKeys.PATH_KEY + " is empty. Expected value: " + this.getPath());
+      return errors;
     }
     if (DatasetDescriptorConfigKeys.DATASET_DESCRIPTOR_CONFIG_ANY.equals(this.getPath())) {
-      return true;
+      return errors;
     }
 
-    if (PathUtils.isGlob(new Path(otherPath))) {
-      return false;
+    if (PathUtils.isGlob(new Path(userFlowConfigPath))) {
+      errors.add(datasetDescriptorPrefix + DatasetDescriptorConfigKeys.PATH_KEY + " is a glob pattern. User input: '" + userFlowConfigPath
+          + "'. Expected input is not of a glob pattern.");
+      return errors;
     }
 
     GlobPattern globPattern = new GlobPattern(this.getPath());
-    return globPattern.matches(otherPath);
+
+    if (!globPattern.matches(userFlowConfigPath)) {
+      errors.add(datasetDescriptorPrefix + ".globPattern is mismatched. User input: '" + userFlowConfigPath
+          + "'. Expected value path of: " + this.getPath() + " and globPattern of '" + globPattern + "'.");
```

Comment:
I don't think .globPattern is a key, it just describes a `DatasetDescriptorConfigKeys.PATH_KEY` that is of glob format e.g. `/path/*`

Semantically the check here checks that if the conf defines that your path needs to be of a certain glob e.g. `/path/*` your input path needs to be contained in said glob.

So the error should look like: DatasetDescriptorPrefix + DatasetDescriptorConfigKeys.PATH_KEY + " user value <userFlowConfigPath> is not contained within the glob of <this.getPath()>. " with proper formatting/templates